### PR TITLE
Code cleanup and move common functions and consts to separate files

### PR DESCRIPTION
**Pull request by tigrazone**
I dont understand why you use hardcoded consts like 0.5f / 180.f * 3.14159f, 45.f / 180.f * 3.14159f and vec3(0.0072f, 1.0f, 0.0034f).
Maybe better to use it as consts?

### CODE REVIEW

**Comment by gameknife** [comment]
Appreciate that! This project is under a time sensitive develop status, some refactor job always delayed...
I'm happy to accept this works, I will merge this after work.

File: assets/shaders/common/VertexFunc.glsl [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
+
+Vertex get_material_data(ivec2 pixel, uvec2 vBuffer, vec3 ray_origin, vec3 ray_direction)
+{
+	// fetch primitive_index high 14bit as instance index and low 18bit as triangle index
+	uint instance_index = vBuffer.x - 1;
```
Comment (gameknife) [inline assets/shaders/common/VertexFunc.glsl:6]: do not -1 here, this leads the hybrid renderer bugs with instance offset
remove it here, and vbuffer.x - 1 outside

File: assets/shaders/ModernDeferredShading.comp
Comment (gameknife) [inline assets/shaders/ModernDeferredShading.comp:1]: vBuffer.x = vBuffer.x - 1; after line 48

**Review by gameknife — CHANGES_REQUESTED** [review]
a small bug when refactor the visibility buffer fetch:

I offset the instance_index in order to leave the default zero value to sky, but the get_material_index function has a slightly different, check the 2 comments.

thanks

File: assets/shaders/common/VertexFunc.glsl (hunk 1 above)
Comment (tigrazone) [reply assets/shaders/common/VertexFunc.glsl:6]: [assets/shaders/common/VertexFunc.glsl](https://github.com/gameknife/gkNextRenderer/pull/2/files/c37f917d6e76756d1b64a4eb6858018d97dcec7e#diff-30597c5a2cbe675c2f0f7fe192aa96f286d0c814f23d6c33b655c62106dc79ae)
line 6 must be
`
	uint instance_index = vBuffer.x;`
?

[assets/shaders/ModernDeferredShading.comp](https://github.com/gameknife/gkNextRenderer/pull/2/files/c37f917d6e76756d1b64a4eb6858018d97dcec7e#diff-fa366bc98c27e2f57423b31a6f71e7f2ec184e3acb03ea6f04eac5f59d8b3df2)
line 48-50 is
```

    uvec2 vBuffer = imageLoad(MiniGBuffer, ipos).rg;
    vec2 uv = vec2(ipos) / vec2(size) * 2.0 - 1.0;
    vec4 origin = Camera.ModelViewInverse * vec4(0, 0, 0, 1);
```

**Comment by tigrazone** [comment]
I will check and fix. Please wait

**Comment by tigrazone** [comment]
done. please check. I use only RayTracingRenderer

**Comment by gameknife** [comment]
PR accepted, 🤝

**Pull request merged**